config/src/test/java/org/springframework/security/config/authentication/JdbcUserServiceBeanDefinitionParserTests.java

```diff
@@ -160,6 +162,35 @@ public void rolePrefixIsUsedWhenSet() {
 		assertThat(AuthorityUtils.authorityListToSet(rod.getAuthorities())).contains("PREFIX_ROLE_SUPERVISOR");
 	}
 
+	@Test
+	public void testEmptyDataSourceRef() {
+		String xml = "<authentication-manager>" + "  <authentication-provider>"
+				+ "    <jdbc-user-service data-source-ref=''/>" + "  </authentication-provider>"
+				+ "</authentication-manager>";
```

Comment:
If you would like to retain the multi-line XML, you can wrap these lines in:

```java
// @formatter:off
...
// @formatter:on
```

I think it would look better than string-concatenating multiple XML elements together on the same line. It would also be consistent with how we do this elsewhere.